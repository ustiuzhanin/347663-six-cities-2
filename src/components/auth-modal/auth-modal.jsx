import React from "react";
import PropTypes from "prop-types";

import { Link } from "react-router-dom";
import { connect } from "react-redux";

import { ActionCreator } from "../../reducer/auth/auth";

const AuthModal = (props) => {
  const { closePopup } = props;
  return (
    <article className="auth-popup">
      <div className="auth-popup__wrapper">
        <button className="auth-popup__btn--close" onClick={() => closePopup()}>
          <span className="visually-hidden">close</span>
        </button>

        <h2 className="auth-popup__title">
          You need to be logged in to use this functionality!
        </h2>

        <Link
          className="auth-popup__btn"
          to={"/login"}
          onClick={() => closePopup()}
        >
          Login
        </Link>
        <Link
          className="auth-popup__btn"
          to={"/signup"}
          onClick={() => closePopup()}
        >
          Sign Up
        </Link>
      </div>
    </article>
  );
};

AuthModal.propTypes = {
  closePopup: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    popupModal: state.auth.popupModal,
  });

const mapDispatchToProps = (dispatch) => ({
  closePopup: () => {
    dispatch(ActionCreator.togglePopup());
  },
});

export { AuthModal };

export default connect(mapStateToProps, mapDispatchToProps)(AuthModal);
