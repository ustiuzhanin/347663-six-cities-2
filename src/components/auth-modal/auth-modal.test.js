import React from "react";
import { AuthModal } from "./auth-modal.jsx";
import ShallowRenderer from "react-test-renderer/shallow";

test(`AuthModal's snapshot`, () => {
  const renderer = new ShallowRenderer();

  renderer.render(<AuthModal closePopup={jest.fn()} />);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
