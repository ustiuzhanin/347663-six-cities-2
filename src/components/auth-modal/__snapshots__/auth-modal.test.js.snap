// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModal's snapshot 1`] = `
<article
  className="auth-popup"
>
  <div
    className="auth-popup__wrapper"
  >
    <button
      className="auth-popup__btn--close"
      onClick={[Function]}
    >
      <span
        className="visually-hidden"
      >
        close
      </span>
    </button>
    <h2
      className="auth-popup__title"
    >
      You need to be logged in to use this functionality!
    </h2>
    <ForwardRef
      className="auth-popup__btn"
      onClick={[Function]}
      to="/login"
    >
      Login
    </ForwardRef>
    <ForwardRef
      className="auth-popup__btn"
      onClick={[Function]}
      to="/signup"
    >
      Sign Up
    </ForwardRef>
  </div>
</article>
`;
